cluster-autoscaler/provisioningrequest/checkcapacity/orchestrator.go

```diff
@@ -0,0 +1,176 @@
+	}
+	if _, err := o.verifyProvisioningRequestClass(unschedulablePods); err != nil {
+		return status.UpdateScaleUpError(&status.ScaleUpStatus{}, errors.NewAutoscalerError(errors.InternalError, err.Error()))
+	}
+
+	o.context.ClusterSnapshot.Fork()
+	defer o.context.ClusterSnapshot.Revert()
+	if err := o.bookCapacity(); err != nil {
+		return nil, errors.NewAutoscalerError(errors.InternalError, err.Error())
+	}
+	scaleUpIsSuccessful, err := o.scaleUp(unschedulablePods)
+	if err != nil {
+		return status.UpdateScaleUpError(&status.ScaleUpStatus{}, errors.NewAutoscalerError(errors.InternalError, "error during ScaleUp: %s", err.Error()))
+	}
+	if scaleUpIsSuccessful {
+		return &status.ScaleUpStatus{Result: status.ScaleUpSuccessful}, nil
+	}
+	return &status.ScaleUpStatus{Result: status.ScaleUpNoOptionsAvailable}, nil
+}
+
+// ScaleUpToNodeGroupMinSize is no-op.
+func (o *provReqOrchestrator) ScaleUpToNodeGroupMinSize(
+	nodes []*apiv1.Node,
+	nodeInfos map[string]*schedulerframework.NodeInfo) (*status.ScaleUpStatus, errors.AutoscalerError) {
+	return nil, nil
+}
+
+func (o *provReqOrchestrator) bookCapacity() error {
+	provReqs, err := o.client.ProvisioningRequests()
+	if err != nil {
+		return fmt.Errorf("Couldn't fetch ProvisioningRequests in the cluster: %v", err)
+	}
+	podsToCreate := []*apiv1.Pod{}
+	for _, provReq := range provReqs {
+		if shouldCapacityBeBooked(provReq) {
+			pods, err := provreq_pods.PodsForProvisioningRequest(provReq)
+			if err != nil {
+				// ClusterAutoscaler was able to create pods before, so we shouldn't have error here.
+				// If there is an error, mark PR as invalid, because we won't be able to book capacity
+				// for it anyway.
+				setCondition(provReq, v1beta1.Failed, metav1.ConditionTrue, "Couldn't book capacity", fmt.Sprintf("couldn't create pods, err: %v", err))
+				continue
+			}
+			podsToCreate = append(podsToCreate, pods...)
+		}
+	}
+	if len(podsToCreate) == 0 {
+		return nil
+	}
+	// scheduling the pods to reserve capacity for provisioning request with BookCapacity condition
+	if _, _, err = o.injector.TrySchedulePods(o.context.ClusterSnapshot, podsToCreate, scheduling.ScheduleAnywhere, false); err != nil {
+		klog.Warningf("Error during capacity booking: %v", err)
+	}
+	return nil
+}
+
+// Assuming that all unschedulable pods comes from one ProvisioningRequest.
+func (o *provReqOrchestrator) scaleUp(unschedulablePods []*apiv1.Pod) (bool, error) {
+	provReq, err := o.client.ProvisioningRequest(unschedulablePods[0].Namespace, unschedulablePods[0].OwnerReferences[0].Name)
+	if err != nil {
+		return false, fmt.Errorf("Failed retrive ProvisioningRequest from unscheduled pods, err: %v", err)
+	}
+	st, _, err := o.injector.TrySchedulePods(o.context.ClusterSnapshot, unschedulablePods, scheduling.ScheduleAnywhere, true)
+	if len(st) < len(unschedulablePods) || err != nil {
+		setCondition(provReq, v1beta1.CapacityFound, metav1.ConditionFalse, "Capacity is not found, CA will try to find later", "")
```

Comment:
Please use the reason as something program-parsable, i.e. `CapacityNotFound` and set the message to something human readable, in this case it can remain `Capacity is not found, CA will try to find later`